pkg/validation/internal/operatorhub.go

```diff
@@ -74,111 +95,301 @@ func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
 		return result
 	}
 
-	errs, warns := validateHubCSVSpec(*bundle.CSV)
-	for _, err := range errs {
+	csvChecksResult := validateHubCSVSpec(*bundle.CSV)
+	for _, err := range csvChecksResult.errs {
 		result.Add(errors.ErrInvalidCSV(err.Error(), bundle.CSV.GetName()))
 	}
-	for _, warn := range warns {
+	for _, warn := range csvChecksResult.warns {
 		result.Add(errors.WarnInvalidCSV(warn.Error(), bundle.CSV.GetName()))
 	}
+
+	errs, warns := validateHubDeprecatedAPIS(bundle, k8sVersion)
+	for _, err := range errs {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+	for _, warn := range warns {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
+	if warn := validateHubChannels(bundle.Channels); warn != nil {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
 	return result
 }
 
-func validateHubCSVSpec(csv v1alpha1.ClusterServiceVersion) ([]error, []error) {
+// validateHubChannels will check the channels. The motivation for the following check is to ensure that operators
+// authors knows if their operator bundles are or not respecting the Naming Convention Rules.
+// However, the operator authors still able to choose the names as please them.
+func validateHubChannels(channels []string) error {
+	const preview = "preview"
+	const stable = "stable"
+	const fast = "fast"
+
+	var channelsNotFollowingConventional []string
+	for _, channel := range channels {
+		if !strings.Contains(channel, preview) &&
+			!strings.Contains(channel, stable) &&
+			!strings.Contains(channel, fast) {
+			channelsNotFollowingConventional = append(channelsNotFollowingConventional, channel)
+		}
+
+	}
+
+	if len(channelsNotFollowingConventional) > 0 {
+		return fmt.Errorf("channel(s) %q are not following the recommended naming convention:"+
+			"https://olm.operatorframework.io/docs/best-practices/channel-naming/", "'"+strings.Join(channelsNotFollowingConventional, `','`)+`'`)
```

Comment:
The `%+q` formatting element will wrap a list of strings in a nicely quoted list-y string.
```suggestion
		return fmt.Errorf("channel(s) %+q are not following the recommended naming convention: "+
			"https://olm.operatorframework.io/docs/best-practices/channel-naming",
			channelsNotFollowingConventional)
```

Reply:
cool